python/arcticdb/version_store/library.py

```diff
@@ -1028,6 +1061,54 @@ def write_metadata(self, symbol: str, metadata: Any) -> VersionedItem:
         """
         return self._nvs.write_metadata(symbol, metadata, prune_previous_version=False)
 
+    def write_metadata_batch(
+        self, payloads: List[WriteMetadataPayload], prune_previous_versions=None
+    ) -> List[VersionedItem]:
+        """
+        Write metadata to multiple symbols in a batch fashion. This is more efficient than making multiple `write` calls
+        in succession as some constant-time operations can be executed only once rather than once for each element of
+        `symbols`.
+        Note that this isn't an atomic operation - it's possible for the metadata for one symbol to be fully written and
+        readable before another symbol.
+        Parameters
+        ----------
+        payloads : `List[WriteMetadataPayload]`
+            Symbols and their corresponding metadata. There must not be any duplicate symbols in `payload`.
+        prune_previous_version : `Optional[bool]`, default=None
+            Remove previous versions from version list. Uses library default if left as None.
+
+        Returns
+        -------
+        List
+            List of versioned items. The data attribute will be None for each versioned item.
+            i-th entry corresponds to i-th element of `symbols`.
+
+        Raises
+        ------
+        ArcticDuplicateSymbolsInBatchException
```

Comment:
What is the behaviour if you try and write metadata for a symbol that doesn't exist?

Reply:
As we discussed --> non-batch version: It creates a symbol if it does not exist. We will leave the behaviour as is, but adding all the logic into the C++ layer. Batch version, we want to do the same that the non-batch version, so we will not raise any exception, breaking the existing API